v2/checksum.js

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+'use strict';
+
+var farm32 = require('farmhash').hash32;
+var crc32 = require('crc').crc32;
+
+var errors = require('./errors');
+var read = require('../lib/read');
+var write = require('../lib/write');
+
+module.exports = Checksum;
+
+// csumtype:1 (csum:4){0,1}
+function Checksum(type, val) {
+    if (!(this instanceof Checksum)) {
+        return new Checksum(type, val);
+    }
+    var self = this;
+    self.type = type;
+    self.val = val || 0;
```

Comment:
The `self.val` field should probably still not default to zero though. Maybe null.